app/internal/restore_events.py

```diff
@@ -0,0 +1,60 @@
+from datetime import datetime, timedelta
+
+from typing import List
+
+from app.database.models import Event, UserEvent
+from app.dependencies import Session
+
+
+def delete_events_after_optionals_num_days(days: int, session: Session):
+    """
+    Delete events permanently after 30 days
+
+    Args:
+        days: number of days to delete from
+        session: db session
+    Returns:
+        None
+    """
+    date_to_delete = datetime.now() - timedelta(days=days)
+
+    user_events_ids_to_be_deleted = (
+        session.query(UserEvent)
+        .join(Event)
+        .filter(Event.deleted_date < date_to_delete)
+        .all()
+    )
+
+    for id_to_be_deleted in user_events_ids_to_be_deleted:
+        (
+            session.query(UserEvent)
+            .filter(UserEvent.id == id_to_be_deleted.id)
+            .delete()
+        )
+
+    session.query(Event).filter(Event.deleted_date < date_to_delete).delete()
+    session.commit()
+
+
+def get_events_ids_to_restored(events_data: List) -> List[str]:
```

Comment:
grammar: get_event_ids_to_restore(), but really can just be get_event_ids()

Reply:
Thanks